conn.go

```diff
@@ -278,6 +278,7 @@ func (c *Conn) Read(b []byte) (int, error) {
 func (c *Conn) ReadMessage(maxBytes int) (Message, error) {
 	batch := c.ReadBatch(1, maxBytes)
 	msg, err := batch.ReadMessage()
+	msg.Watermark = batch.Watermark()
```

Comment:
I'm not sure about this. Maybe we want to have `Watermark` methods similar to `Offset`?

maybe something like is better:

```
for {
  msg := stream.Read()
  stat.Gauge("watermark", stream.Watermark())
}
```